kola/tests/rpmostree/deployments.go

```diff
@@ -0,0 +1,300 @@
+        },
+        "mode": 420
+      }
+    ]
+  }
+}`),
+
+		Distros:          []string{"rhcos"},
+		ExcludePlatforms: []string{"qemu"}, //these need network to retrieve bits
+	})
+}
+
+// rpmOstreeUpgradeRollback simulates an upgrade by creating a local branch, making
+// a commit to the branch, and upgrading the host to said commit.  After a successful
+// "upgrade", the host is rolled back to the original deployment.
+func rpmOstreeUpgradeRollback(c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	originalStatus, err := getRpmOstreeStatusJSON(c, m)
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// create a local branch to act as our upgrade target
+	originalOrigin := originalStatus.Deployments[0].Origin
+	originalCsum := originalStatus.Deployments[0].Checksum
+	createBranch := "sudo ostree refs --create local-branch " + originalCsum
+	c.MustSSH(m, createBranch)
+
+	// make a commit to the new branch
+	refspec := strings.Split(originalStatus.Deployments[0].Origin, ":")[1]
+	createCommit := "sudo ostree commit -b local-branch --tree ref=" + refspec + " --add-metadata-string version=kola-test-1.0"
+	newCommit := c.MustSSH(m, createCommit)
+
+	// edit origin file to point to new branch
+	currentDir := c.MustSSH(m, "sudo ostree admin --print-current-dir")
+	originFile := string(currentDir) + ".origin"
+	sedCmd := "sudo sed -i 's|^refspec=.*$|refspec=local-branch|' " + originFile
+	c.MustSSH(m, sedCmd)
+
+	// use rpm-ostree upgrade to get to the "new" commit
+	c.MustSSH(m, "sudo rpm-ostree upgrade")
```

Comment:
I was emulating what we did in a-h-t; I guess if the end result is the same I could use `rebase`, but I was trying to tickle the `upgrade` code path.

Reply:
Understood, but internally rpm-ostree is really doing almost exactly the same thing.  `upgrade`  along with most other verbs get unrolled into the generic `UpdateDeployment` DBus API.

One thing missing with this "sed behind rpm-ostree's back" approach is that it might be racy without `rpm-ostree reload`.  I also consider the `.origin` file format mostly private to ostree - in theory we could change, though it is unlikely.

Reply:
(Although there are differences between `upgrade` and `rebase` - one of those is that the latter doesn't check that the timestamp on the commit is newer, but we're not testing that here)

Reply:
OK, since it's basically the same, I'll go with `rebase`.  Thanks for the additional details!